lib/OpenQA/WebAPI/Controller/API/V1/Search.pm

```diff
@@ -123,27 +111,54 @@ sub query {
             }
             $last_filename = $filename;
             push(@results, {occurrence => "$basename/$filename", contents => $contents});
+            $cap--;
         }
-        $cap -= scalar @results;
         last if $cap < 1;
+    }
+    return @results;
```

Comment:
@kraih @perlpunk Would it be more performant to return this possible big array as arrayref or does Perl something similar to C++'s copy elision?

Reply:
Likely the most efficient way would be passing an arrayref to `@results` directly to the functions so they'd push directly to `@results` instead of using intermediate arrays and combining them later. It would likely be even less code.